server/grpc_service.go

```diff
@@ -244,36 +245,40 @@ func (s *Server) RegionHeartbeat(server pdpb.PD_RegionHeartbeatServer) error {
 			return errors.Trace(err)
 		}
 
-		if err = s.validateRequest(request.GetHeader()); err != nil {
-			// TODO: How to close this stream?
-			return errors.Trace(err)
-		}
-
 		cluster := s.GetRaftCluster()
 		if cluster == nil {
 			msg := "cluster is not bootstrapped"
-			err = sendErrorRegionHeartbeatResponse(server, s.clusterID, pdpb.ErrorType_NOT_BOOTSTRAPPED, msg)
+			err = regionHeartbeatErr(server, s.clusterID, pdpb.ErrorType_NOT_BOOTSTRAPPED, msg)
 			if err != nil {
 				return errors.Trace(err)
 			}
 			continue
 		}
 
+		if isNew {
+			if err = s.validateRequest(request.GetHeader()); err != nil {
+				return errors.Trace(err)
+			}
+			storeID := request.GetLeader().GetStoreId()
+			cluster.coordinator.hbStreams.bindStream(storeID, server)
```

Comment:
Is it safe to send with `server` concurrently?
Seems `server` will still be used in this function.